File: src/services/actividad.service.ts
/**
 * Servicio de documentos de actividad.
 * Listar y descargar documentos asociados a farmacias.
 */
import apiClient from './api-client';
import { Documento } from '../types';

export const actividadService = {
  /**
   * Listar documentos de actividad de una farmacia.
   */
  async listar(farmaciaId: number): Promise<Documento[]> {
    const response = await apiClient.get<{ data: Documento[] }>(
      `/farmacias/${farmaciaId}/documentos`
    );
    return response.data.data;
  },

  /**
   * Obtener URL de descarga de un documento.
   */
  getDownloadUrl(documentoId: number): string {
    return `${apiClient.defaults.baseURL}/documentos/${documentoId}/download`;
  },
};

File: lib/services/actividad_service.dart
import '../core/api_client.dart';
import '../core/secure_storage.dart';
import '../models/documento.dart';

/// Servicio de documentos de actividad.
class ActividadService {
  /// Listar documentos de actividad + documentación del usuario.
  static Future<List<Documento>> listar() async {
    final response = await apiClient.get('/documentos/list');
    final data = response.data['data'] as List<dynamic>;
    return data.map((e) => Documento.fromJson(e as Map<String, dynamic>)).toList();
  }

  /// Obtener URL completa de descarga de un documento (incluye token para auth).
  static Future<String> getDownloadUrl(int documentoId, {String fuente = 'actividad'}) async {
    final token = await SecureStorage.getToken();
    return '$apiBaseUrl/documentos/download?fuente=$fuente&id=$documentoId&token=$token';
  }
}
